### ISSUE TITLE
407 Proxy Authentication Required error when running gsutil on Cloud SDK 296.0.0 or above using a proxy

### ISSUE THREAD

**Issue by turribeach** [GoogleCloudPlatform/gsutil#1065]
Running "gsutil" on a Windows PC configured to use a proxy returns the "407 Proxy Authentication Required" despite all proxy settings being correctly set. I traced the problem to an update on Cloud SDK v296.0.0:

[https://cloud.google.com/sdk/docs/release-notes#29600_2020-06-09](https://cloud.google.com/sdk/docs/release-notes#29600_2020-06-09)

It seems to be the "Updated bundled Python executable to 3.7.5." line. In version 295.0.0 the bundled Python is v2.7.13 so gsutil using a proxy seems to break on Python 3.7.5.

To verify this issue:

1. Run "gcloud --version" to verify your Google Cloud SDK is 295.0.0 or earlier
2. Then confirm "gsutil ls" runs fine using a proxy
3. Then run "gcloud components update" to update your Google Cloud SDK to the latest version (likely to be 300.0.0 or above)
4. Now run "gsutil ls" again and you should get a "407 Proxy Authentication" error despite proxy settings being correctly set

You can rollback to version 295.0.0 using this command: "gcloud components update --version 295.0.0"

You can also reproduce the problem setting the CLOUDSDK_GSUTIL_PYTHON variable to use a python3 interpreter like this:

`set CLOUDSDK_GSUTIL_PYTHON=c:\[path to your python3 installation\python`

As a workaround I was able to update the Google Cloud SDK to the current latest version (300.0.0) and set CLOUDSDK_GSUTIL_PYTHON to point to the embedded Python 2 interpreter which allows gsutil to work fine using a proxy:

`set CLOUDSDK_GSUTIL_PYTHON=C:\Dev\Programs\google-cloud-sdk\platform\bundledpython2\python`

However Google claims on the [Google Cloud SDK release notes](https://cloud.google.com/sdk/docs/release-notes) that support for Python 2 will be removed on July 31st 2020 so presumably this work around will stop working when they remove Python 2 from the Google Cloud SDK.

**Comment by turribeach** [comment]
We are seeing more people in our organisation find this problem, as they update their Google Cloud SDK. A couple of points worth mentioning:

1. The workaround above using CLOUDSDK_GSUTIL_PYTHON to point to a v2 Python interpreter only works if you have the Python embedded Cloud SDK version. This Cloud SDK version is only available for Windows. If you are on Linux you will need to point CLOUDSDK_GSUTIL_PYTHON to some other Python 2 installation to work around the problem. Most Linux versions do have a Python 2 interpreter, sample for Linux "export CLOUDSDK_GSUTIL_PYTHON=/usr/bin/python2"
2. If you use the non-embedded Google Cloud SDK on Windows or any Google Cloud SDK version on Linux with a Python 3 interpreter you will see the issue I described above, even if you are on Google Cloud SDK earlier than v296.0.0.

I have raised a support case with Google (Case#24292113) and I will post an update here once Google responds. But if you are having this issue it will be good to know to confirm it's impacting other people.
